File: code/functions/recursion.cpp
#include &lt;iostream&gt;

// Recursion in C++
int factorial(int n) {
    return n &lt;= 1 ? 1 : n * factorial(n - 1);
}

int main() {
    std::cout &lt;&lt; factorial(5) &lt;&lt; std::endl;
    return 0;
}

File: code/functions/recursion.erl
% Recursion in Erlang
-module(math).
-export([factorial/1, start/0]).

factorial(0) -&gt; 1;
factorial(N) -&gt; N * factorial(N - 1).

start() -&gt;
    io:format("~p~n", [factorial(5)]).
